From 33dd8ac3a1513fda0c0c3ea3e8b7e10e23b7c795 Mon Sep 17 00:00:00 2001
From: "rajeshgopu.iv" <rajeshgopu.iv@lge.com>
Date: Thu, 30 Nov 2017 18:18:44 +0530
Subject: [PATCH] apollo empty launcher no space db8 fix

:Release Notes:
empty launcheR on db8 no space sceanrio fix

:Detailed Notes:
when /var/db is full ,db8 systemdb script does
factory resets maindb after removing current maindb
because of not factory reset api it was with empty
maindb & this caused empty launcher apps.
Fix is to remove factory reset command & add below
rm -rf /var/spool/* ,clear space on /var/
rm -rf /var/cache/configurator/* ,to remove configurator caches
reboot ,to reboot to recreate db entries by all apps

:Testing Performed:
Tested locally

:QA Notes:

:Issues Addressed:
[PLAT-45863] [apollo] surface-manager crashes in sequences of app
             launching -- empty launcher after reboo
---
 files/scripts/errorNoSpace.bash    | 4 +++-
 files/scripts/errorOpenMainDb.bash | 6 +++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/files/scripts/errorNoSpace.bash b/files/scripts/errorNoSpace.bash
index ef490db..ae240aa 100755
--- a/files/scripts/errorNoSpace.bash
+++ b/files/scripts/errorNoSpace.bash
@@ -7,7 +7,9 @@ mkdir -p /mnt/lg/cmn_data/db8 || true
 touch /mnt/lg/cmn_data/db8/errorNoSpace || true
 rm -f /var/luna/preferences/ran-firstuse || true
 rm -rf /var/db/main/* || true
+rm -rf /var/cache/configurator/* || true
+rm -rf /var/spool/* || true
 sync
 
 PmLogCtl log DB8 crit "mojodb-luna [] DB8 DBGMSG {} [upstart_maindb] No space left for maindb"
-/usr/bin/luna-send -n 1 luna://com.webos.service.tv.systemproperty/doUserDefault '{}'
+reboot
diff --git a/files/scripts/errorOpenMainDb.bash b/files/scripts/errorOpenMainDb.bash
index 2f53e7d..30db385 100755
--- a/files/scripts/errorOpenMainDb.bash
+++ b/files/scripts/errorOpenMainDb.bash
@@ -19,14 +19,18 @@ if [ "$BYTES_FREE" -ge "$BYTES_THRESHOLD" ]; then
     tar cjf $BACKUP_FILENAME /var/db/main
     chmod 400 $BACKUP_FILENAME
     rm -rf /var/db/main/*
+    rm -rf /var/cache/configurator/* || true
+    rm -rf /var/spool/* || true
     rm -f /var/luna/preferences/ran-firstuse || true
 else
     mkdir -p /mnt/lg/cmn_data/db8 || true
     touch /mnt/lg/cmn_data/db8/errorOpenMainDb || true
     rm -rf /var/db/main/* || true
+    rm -rf /var/cache/configurator/* || true
+    rm -rf /var/spool/* || true
     rm -f /var/luna/preferences/ran-firstuse || true
     PmLogCtl log DB8 crit "mojodb-luna [] DB8 DBGMSG {} [maindb_errorOpenmainDb.bash] No space left to store corrupted maindb"
 fi
 
 #do factory reset
-/usr/bin/luna-send -n 1 luna://com.webos.service.tv.systemproperty/doUserDefault '{}'
+reboot
-- 
1.9.1

